lightning/src/ln/channelmanager.rs

```diff
@@ -6034,7 +6034,12 @@ impl<Signer: Sign, M: Deref , T: Deref , K: Deref , F: Deref , L: Deref >
 		}
 	}
 
-	fn peer_connected(&self, counterparty_node_id: &PublicKey, init_msg: &msgs::Init) {
+	fn peer_connected(&self, counterparty_node_id: &PublicKey, init_msg: &msgs::Init) -> Result<(), ()> {
+		if !init_msg.features.supports_static_remote_key() {
```

Comment:
It's not obvious to me why we wouldn't also disallow peers that don't support `payment_secret`, since it looks like we'll fail payments that don't have one.